Improve npm package Windows installation robustness

rishiyaduwanshi opened the pull request:
This PR builds upon the existing fix in #453 by adding additional robustness improvements for Windows npm package installation:

**Enhancements:**
- Enhanced PowerShell command flags: -NoProfile -ExecutionPolicy Bypass -LiteralPath
- Added 100ms delay after file.close() to ensure file handles are released
- Improved path handling with absolute paths and proper escaping
- Added try-catch-finally for better cleanup
- windowsHide flag to prevent PowerShell window flashes
- Cleanup of partial files on redirects
- Better error messages for debugging

**Changes:**
- Reorganized npm package structure into dedicated npm-package/ folder
- Updated workflows to use template-based package.json generation
- Added testing scripts for local validation (test-local.ps1, test-local.sh)
- Added CI workflow for multi-platform testing

**Testing:**
Tested on Windows 11 with PowerShell 5.1. The installation now completes successfully without ENOENT errors.

Closes #453

sinelaw commented on npm-package/binary-install.js:
```diff
@@ -8,79 +8,125 @@ const VERSION = require('./package.json').version;
+                    if (err) reject(err);
+                    else if (process.platform === 'win32') {
+                        // Wait briefly on Windows to ensure file handle is fully released
+                        setTimeout(resolve, 100);
```
✦ setTimeout(resolve, 100) is a fragile "magic number" fix for Windows file locking. A more robust approach is to use a retry loop around the extraction:

    1 // Robust alternative to setTimeout
    2 async function withRetry(fn, retries = 5) {
    3   for (let i = 0; i < retries; i++) {
    4     try { return fn(); }
    5     catch (e) { 
    6       if (i === retries - 1) throw e;
    7       await new Promise(r => setTimeout(r, 100 * (i + 1))); 
    8     }
    9   }
   10 }

  This ensures the installation succeeds even on slower systems or under heavy antivirus scanning.

sinelaw commented:
@Rishiyaduwanshi please take a look at the run - it created many jobs called Test npm Package and they failed (can we have a single job + not fail?)

Rishiyaduwanshi commented:
@sinelaw I have removed now  `test-npm-package.yml` workflow entirely because:

1. It was creating 9 jobs (3 OS × 3 Node versions) - excessive for simple template validation
2. It's redundant - `release.yml` and `release-npm.yml`already tests npm package during actual releases
3. Added `npm-package/local.test.js` for local developer testing instead
4. The old workflow was also failing because it missed copying README during testing

The new `local.test.js` script:
- Tests with latest GitHub release version (or specify version manually)
- Supports `--keep` flag for manual inspection/npm link

Rishiyaduwanshi commented:
@sinelaw waiting for workflow approval
